test/hotspot/jtreg/compiler/c2/irTests/UDivINodeIdealizationTests.java

```diff
@@ -0,0 +1,148 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.irTests;
+
+import jdk.test.lib.Asserts;
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8332268
+ * @summary Test that Ideal transformations of UDivINode* are being performed as expected.
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.UDivINodeIdealizationTests
+ */
+public class UDivINodeIdealizationTests {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Run(test = {"constant", "identity", "identityAgain", "identityAgainButBig", "identityThird",
+            "retainDenominator", "divByPow2", "divByPow2Big" })
+    public void runMethod() {
+        int a = RunInfo.getRandom().nextInt();
+        a = (a == 0) ? 1 : a;
+        int b = RunInfo.getRandom().nextInt();
+        b = (b == 0) ? 1 : b;
+
+        int min = Integer.MIN_VALUE;
+        int max = Integer.MAX_VALUE;
+
+        assertResult(0, 0, true);
+        assertResult(a, b, false);
+        assertResult(min, min, false);
+        assertResult(max, max, false);
+    }
+
+    @DontCompile
+    public void assertResult(int a, int b, boolean shouldThrow) {
+        try {
+            Asserts.assertEQ(Integer.divideUnsigned(a, a), constant(a));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        try {
+            Asserts.assertEQ(Integer.divideUnsigned(a, Integer.divideUnsigned(b, b)), identityThird(a, b));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        try {
+            Asserts.assertEQ(Integer.divideUnsigned((a * b), b), retainDenominator(a, b));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        Asserts.assertEQ(Integer.divideUnsigned(a, 1), identity(a));
+        Asserts.assertEQ(Integer.divideUnsigned(a, Integer.divideUnsigned(13, 13)), identityAgain(a));
+        Asserts.assertEQ(Integer.divideUnsigned(a, 8), divByPow2(a), "divByPow2 " + a);
+        Asserts.assertEQ(Integer.divideUnsigned(a, -2147483648), divByPow2Big(a));
+        Asserts.assertEQ(Integer.divideUnsigned(a, Integer.divideUnsigned(-2129457054, -2129457054)), identityAgainButBig(a));
+    }
+
+    @Test
+    @IR(failOn = {IRNode.DIV})
```

Comment:
All these should be `failOn = {IRNode.UDIV}`

Reply:
Good catch, thanks! Fixed it.